core/src/main/scala/akka/persistence/query/scaladsl/EventsBySliceQuery.scala

```diff
@@ -19,7 +19,22 @@ import akka.stream.scaladsl.Source
  * `EventsBySliceQuery` that is using a timestamp based offset should also implement [[EventTimestampQuery]] and
  * [[LoadEventQuery]].
  */
-trait EventsBySliceQuery extends ReadJournal {
+trait EventsBySliceQuery[Event] extends ReadJournal {
+
+  // FIXME should we have have the Event type parameter here or on the `def eventsBySlices`.
+  // For example the new DurableState queries have it here.
+  // One drawback of having it here is that this is not possible:
+  //   private val query = PersistenceQuery(testKit.system).readJournalFor[R2dbcReadJournal](R2dbcReadJournal.Identifier)
+  //   query.eventsBySlices(...)
+  // one would have to cast it to more specific EventsBySliceQuery[TheEvent]
+  // or use that in the readJournalFor class
+  // on the other hand, the alternative is also defining the type:
+  // query.eventsBySlices[TheEvent](...)
+  //
+  // Maybe it should be on the def because the same R2dbcReadJournal instance can be used for different entityTypes,
+  // i.e. different Event classes
+  //
+  // The original reason I changed it was some trouble with similar param in LoadEventQuery
```

Comment:
I'm on the fence here and would like input/opinions

Reply:
I think it should be on the method for the reasons you mentioned. We can store different events types.

I don't get which kind of trouble you had with `LoadEventQuery`. Could you clarify?

Reply:
Thanks, I'll try with that again, first with LoadEventQuery.

Reply:
done in https://github.com/akka/akka-persistence-r2dbc/pull/144/commits/0b3e9ea4030ab29a38c7e96356f876c2a1cecf2d